### ISSUE TITLE
The number of TS of link capacities has an influence on the draw of thermal availabilities

### ISSUE THREAD

**Issue by PPlessiez** [AntaresSimulatorTeam/Antares_Simulator#1666]
**Description**
On a given study, the number of input TS for link capacities can have an influence on the draw of the TS of thermal availabilities.
More specifically, if there is only one TS for link capacities, the draw of thermal availabilities will be different from the case in which there are several TS for link capacities. As long as there are several TS for link capacities (2, 3, N), the draw of thermal availabilities should be identical.

**How to Reproduce**
See the attached UC: when changing the number of TS for link capacities from 1 to N (N >1), the draw of thermal availabilities in the "production" node will be different (can be checked in the "ts-number" output directory).

**Expected behavior**
The number of TS of any input data should not have an influence on any random TS draws. Only the number of areas or the number of clusters can change these draws.

**Screenshots and additional files**
[test_5TS_github.zip](https://github.com/AntaresSimulatorTeam/Antares_Simulator/files/12783423/test_5TS_github.zip)

**OS and version**
 - OS: Windows
 - Antares Simulator version: 8.5

**Comment by flomnes** [comment]
This is completely normal, since the random generator is shared among links, thermal clusters, etc.

In pseudo-code

```cpp
for (area: areas) {
    for (cluster: area.clusters) {
        cluster.tsNumber = random.next();
    }
    for (link: area.links) {
        if (numberOfTS == 1) {
            link.tsNumber = 1; // For compatibility reasons
        } else {
            link.tsNumber = random.next();
        }
    }
}
```

**Comment by PPlessiez** [comment]
To sum up our discussions:
- The special treatment on links was added to ensure that results of already existing studies with new Antares versions would keep the same random draws. But the consequence is that if, for a given link on the study, the user changes from 1 TS to several, all random draws are changed.
- If we changed this special case to ensure that the number of TS on a link has no influence on random draws, the retrocompatibility would not be ensured anymore.
